controlplane/pkg/nsmd/nsmd_crossconnect_client.go

```diff
@@ -185,7 +185,13 @@ func (client *NsmMonitorCrossConnectClient) dataplaneCrossConnectMonitor(datapla
 
 func (client *NsmMonitorCrossConnectClient) remotePeerConnectionMonitor(remotePeer *registry.NetworkServiceManager, ctx context.Context) {
 	logrus.Infof("Connecting to Remote NSM: %s", remotePeer.Name)
-	conn, err := grpc.Dial(remotePeer.Url, grpc.WithInsecure())
+	tracer := opentracing.GlobalTracer()
+	conn, err := grpc.Dial(remotePeer.Url, grpc.WithInsecure(),
+		grpc.WithUnaryInterceptor(
+			otgrpc.OpenTracingClientInterceptor(tracer)),
+		grpc.WithStreamInterceptor(
+			otgrpc.OpenTracingStreamClientInterceptor(tracer)))
+
```

Comment:
@platoff Is the issue that jaeger doesn't report until the ultimate parent span is closed... and since the ultimate parent span is on the grpc call, it never gets closed?